### ISSUE TITLE
Conventional Commits pre-commit check is not working

### ISSUE THREAD

**Issue by srid** [juspay/services-flake#130]
The second commit in PR #127 has the commit message [use same installer](https://github.com/juspay/services-flake/pull/127/commits/dc1c67fd0addf1b20ddd55f4b0631db67431f50b) - yet the pre-commit check did not catch it, as it doesn't comfort to conventional commits spec.

**Comment by srid** [comment]
Commit's vanished because I had to force push, but the commit message was literally `use same installer`.

**Comment by srid** [comment]
The hook itself works,

<img width="904" alt="image" src="https://github.com/juspay/services-flake/assets/3998/2daf0386-422a-45c5-a8fe-90e492b19693">

However, if I skip it:

<img width="529" alt="image" src="https://github.com/juspay/services-flake/assets/3998/82f0a494-5497-402b-a151-41a646c3bbfa">

And run the flake check:

<img width="611" alt="image" src="https://github.com/juspay/services-flake/assets/3998/35686166-a4ae-41cd-b495-7c5c038748f6">

<img width="675" alt="image" src="https://github.com/juspay/services-flake/assets/3998/6447a9ca-ec71-48f3-a0d0-d263d039ef50">

... it passes ^

**Comment by srid** [comment]
I don't think the pre-commit module actually adds a flake check. 

Also, https://github.com/convco/convco

We can run this manually

<img width="1102" alt="image" src="https://github.com/juspay/services-flake/assets/3998/2352e47a-06b8-4d43-9ffa-088a92f041d2">

**Comment by terlar** [comment]
Yeah, when running in a flake check it won't have access to `.git/COMMIT_EDITMSG` and will therefore not be able to check the commit message. The workaround is to use pre-commit directly on CI. That is what I found as well, although I was using conform.

**Comment by srid** [comment]
A PR can have multiple commits, so whatever flake check add must check all of them.

**Comment by srid** [comment]
But git history will be available, perhaps it should be run outside of Nix, in github actions.

**Comment by shivaraj-bh** [comment]
This commit checks for all the commits on the current branch, starting from `main`:

```sh
cz check --rev-range main..HEAD
```

**Comment by srid** [comment]
Cool. We should also consider #134 when deciding between `cz` and `convco`.

**Comment by shivaraj-bh** [comment]
> This command checks for all the commits on the current branch, starting from `main`:
> 
> ```shell
> cz check --rev-range main..HEAD
> ```

Can’t do this until https://github.com/NixOS/nix/issues/6900

**Comment by shivaraj-bh** [comment]
I have made some progress at https://github.com/juspay/services-flake/tree/cz-check, will wait until there is a reliable way to copy `.git` to the store.

**Comment by srid** [comment]
Copying `.git` to the Nix store sounds like a bad idea in general? We only need history of commit messages, not the whole index (much less copy a variation of it every time to the store).

**Comment by shivaraj-bh** [comment]
> We only need history of commit messages, not the whole index

I doubt we can pass this to flake check in a straightforward way.

One hack to make this work will be to use `writeShellApplication` and run `cz check --rev-range main..HEAD` via `nix run`.

**Comment by srid** [comment]
FYI: this recent commit https://github.com/juspay/services-flake/commit/5960a2c553ba6f5efec8d6aad64219c8fa036e50 does not follow the convention, so it is going to be missed when we next generate the changelog. @shivaraj-bh
